plugin-core/resources/bzleclipse_aspect.bzl

```diff
@@ -63,10 +73,23 @@ def java_rule_ide_info(target, ctx):
   else:
      sources = []
 
-  jars = [library_artifact(output) for output in target[JavaInfo].outputs.jars]
-  classpath_jars = depset([jar
+  jars = []
+  classpath_jars = depset()
+  if target[JavaInfo].outputs.jars != None and len(target[JavaInfo].outputs.jars) > 0:
+    # standard Java library or test rule
+    jars = [library_artifact(target, output) for output in target[JavaInfo].outputs.jars]
+    classpath_jars = depset([jar
        for output in target[JavaInfo].outputs.jars
        for jar in jars_from_output(output)])
+  else:
+    # proto-java library rules end up in here, no jars listed on the output object, so we resort to using transitive_runtime_deps
```

Comment:
java_proto_library does not put the generated jar file in the output provider as you would expect, so we have to fall back and add all transitive runtime jars if outputs are empty.